pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -431,20 +431,30 @@ func (p *adProvider) searchPrincipals(name, principalType string, config *v3.Act
 }
 
 func (p *adProvider) searchUser(name string, config *v3.ActiveDirectoryConfig, caPool *x509.CertPool) ([]v3.Principal, error) {
-	srchAttributes := strings.Split(config.UserSearchAttribute, "|")
-	query := "(&(" + ObjectClassAttribute + "=" + config.UserObjectClass + ")"
-	srchAttrs := "(|"
-	for _, attr := range srchAttributes {
-		srchAttrs += "(" + attr + "=" + name + "*)"
+	var query string
+	if config.UserSearchQuery != "" {
+		query = config.UserSearchQuery
```

Comment:
The query should not function as a replacement to our normal UserSearchAttribute based search; it should be an additional filter to the search.
So, you should end up withe a search that looks like this (excuse my bad pseudo-code) assuming you are search for someone with the last name smith:
```
(&(objectClass=person)(|((sAMAccountName=smith)(sn=smith)(givenName=smith))(<user entered search query))
```

The success criteria should be that if you have two different users where sn = smith but differ in some other way, you can add a custom query filter that would cause rancher to just return one of the two smiths